### ISSUE TITLE
SDK deprecation

### ISSUE THREAD

**Issue by CharlesIC** [mosen/mysides#14]
Many SDKs used in the project (e.g. the _LSSharedFileListCreate_ method) have been deprecated in macOS 10.12 (Sierra). They still work (for now?) but it's worth updating to the new Service Management framework. More info at these links:
[OS X at startup" code deprecation](https://github.com/bitcoin/bitcoin/issues/8420)
[Make Swift Cocoa app launch on startup on OS X 10.11](https://stackoverflow.com/questions/35339277/make-swift-cocoa-app-launch-on-startup-on-os-x-10-11/35356972#35356972)

**Comment by mosen** [comment]
Yeah many of the shared file list functions are going behind XPC so it's sfltool or nothing unfortunately.

**Comment by CharlesIC** [comment]
OK, I wasn't able to find a new way of manipulating the SFL files that is meant to replace the deprecated methods in LSSharedFileList.h. Unless you need to remove an item, sfltool will do, otherwise the current method seems to be the way to go at least until the functionality gets removed from the OS.

**Comment by mosen** [comment]
I'll try to update it but otherwise we need a new method, you are correct

**Comment by wongzigii** [comment]
Any update on this? Is there any replacement for the deprecated LSSharedFileList?

**Comment by mosen** [comment]
Nope. There's API private to apple but we can't use it

**Comment by RicardoAnjos** [comment]
In Mojave LSSharedFileListCreate and LSSharedFileListItemRemove are no longer. Only listing. So the last version were working is in High Sierra. Would be great to find another way to add and remove icons from favourites though

**Comment by Tatsh** [comment]
Not having a problem with this in Mojave @RicardoAnjos.
